test/differentials/composite.jl

```diff
@@ -56,6 +60,10 @@ end
         @test getproperty(Composite{Tuple{Float64,}}(@thunk 2.0^2), 1) == 4.0
         @test getproperty(Composite{Tuple{Float64,}}(a=(@thunk 2.0^2),), :a) == 4.0
 
+        # TODO: uncomment this once https://github.com/JuliaLang/julia/issues/35516
+        # @test haskey(Composite{Tuple{Float64}}(2.0), 1) == true
+        # @test hasproperty(Composite{Tuple{Float64}}(2.0), 2) == false
```

Comment:
Would it be better to use `test_broken`, so we get notified when this is starts working?